app/javascript/components/textual_summary/table_list_view.jsx

```diff
@@ -28,15 +28,25 @@ const renderRow = (row, i, colOrder, rowLabel, onClick) => (
 export default function TableListView(props) {
   const { headers, values, title } = props;
 
+  /** Function to generate headers for structured list. */
+  const miqListHeaders = (list) => (!!list[0].key ? list.map((item) => item.label) : list);
+
   /** Function to generate rows for structured list. */
   const miqListRows = (list) => {
-    const headerKeys = props.headers.map((item) => (item.key));
-    return list.map((item) => (headerKeys.map((header) => item[header])));
+    // For More Complicated Tables, e.g. Tenant Quotas
+    if (!!headers[0].key) {
+      const headerKeys = headers.map((item) => (item.key));
+      return list.map((item) => (headerKeys.map((header) => item[header])));
+    }
+
+    const data = [];
+    list.map((item) => data.push({ ...item, label: item.name }));
+    return data;
```

Comment:
should this be an if/else?  

```js
    if (!!headers[0].key) {
      const headerKeys = headers.map((item) => (item.key));
      return list.map((item) => (headerKeys.map((header) => item[header])));
    } else {
      const data = [];
      list.map((item) => data.push({ ...item, label: item.name }));
      return data;
    }
```

Are these worthy of named functions?  `extractRowDataWithHeaderBasedMapping` and `extractRowDataWithSimpleLabels`. Basically, how can we make it easier to know which controller tables would fall into the correct if/else logic.

Reply:
Made some changes to hopefully make things clearer, let me know how you feel about them

Reply:
It looks much clearer to me, thanks!  Can @GilbertCherrie or @jeffibm review it?

Reply:
Hey @DavidResende0 , you are right. I didn't think of translations :)

we will go ahead with your changes.